### PR TITLE
fix: Fix usage of deprecated APIs

### LINKED ISSUE #732: Fix usage of deprecated APIs
[kind/Enhancement]

OpenShift Toolkit by Red Hat 1.8.0 uses deprecated API, which may be removed in future releases leading to binary and source code incompatibilities

Screenshot 2024-03-04 at 6 36 54 PM

### PR DESCRIPTION
**Pull request by sbouchet**
## Was the change discussed in an issue?
fixes (partially) #732

### CODE REVIEW

**Review by adietish — APPROVED** [review]
LGTM

File: src/main/java/org/jboss/tools/intellij/openshift/utils/odo/JSonParser.java [hunk 1]
```diff
@@ -12,251 +12,241 @@
...
-        if (featuresNode.has(ComponentFeature.Mode.DEBUG_MODE.getLabel().toLowerCase()) && featuresNode.get(ComponentFeature.Mode.DEBUG_MODE.getLabel().toLowerCase()).asBoolean()) {
-            result.add(ComponentFeature.Mode.DEBUG_MODE);
-        }
-        if (featuresNode.has(ComponentFeature.Mode.DEPLOY_MODE.getLabel().toLowerCase()) && featuresNode.get(ComponentFeature.Mode.DEPLOY_MODE.getLabel().toLowerCase()).asBoolean()) {
-            result.add(ComponentFeature.Mode.DEPLOY_MODE);
-        }
-        return result;
-    }
-
-    public DebugStatus parseDebugStatus() {
```
Comment (adietish) [inline src/main/java/org/jboss/tools/intellij/openshift/utils/odo/JSonParser.java:164]: Afaics this method was never used and could therefore be eliminated +1. Afaiu this has the benefit that apache commons is not used any more.

Comment (sbouchet) [reply src/main/java/org/jboss/tools/intellij/openshift/utils/odo/JSonParser.java:164]: i'm looking into other places that *still* uses apache Stringutils class

**Pull request merged**